File: CommandPython/Stereo.py
# Copyright (c) 2022 RodKingroo

class Stereo:
    def __init__(self, ui):
        self.UNKNOWN = 0
        self.CASSETTE = 1
        self.CD = 2
        self.DVD = 3
        self.RADIO = 4
        self.CLOUDMUSIC = 5
        self.volume = 0
        self.state = self.UNKNOWN
        self.ui = ui
        
    def on(self):
        self.ui.set_text_output('Stereo On')
        
    def off(self):
        self.ui.set_text_output('Stereo Off')
    
    #setted cassete
    def setCASSETTE(self):
        self.state = self.CASSETTE
        self.ui.set_text_output('Cassette Rec selected')
    
    #setted CD
    def setCD(self):
        self.state = self.CD
        self.ui.set_text_output('CD device selected')
    
    #setted DVD
    def setDVD(self):
        self.state = self.DVD
        self.ui.set_text_output('DVD device selected')
        
    #setted Radio Example: play Radio Station
    def setRADIO(self):
        self.state = self.RADIO
        self.ui.set_text_output('Radio selected')
        
    # setted cloud music Example: Spotify, Yandex Music
    def setCLOUDMUSIC(self):
        self.state = self.CLOUDMUSIC
        self.ui.set_text_output('Voice Assist play music')
    
    def setVolume(self, volume):
        self.volume = volume
        self.ui.set_text_output("Volume level: " + str(self.volume))
        
    def getVolume(self):
        return self.volume
    
    def getState(self):
        return self.state
    

File: CommandCPP/Stereo.cpp
/* Copyright (c) 2022 RodKingroo */
#include "Stereo.h"

void Stereo::on()
{
    cout << "Stereo On" << endl;
}

void Stereo::off()
{
    cout << "Stereo Off" << endl;
}


void Stereo::setCassette()
{
    state = CASSETTE;
    cout << "Cassette Rec Selected" << endl;
}

void Stereo::setCD()
{
    state = CD;
    cout << "CD device selected" << endl;
}

void Stereo::setDVD()
{
    state = DVD;
    cout << "DVD device selected" << endl;
}

void Stereo::setRadio()
{
    state = RADIO;
    cout << "Radio device selected" << endl;
}

void Stereo::setCloud()
{
    state = CLOUDMUSIC;
    cout << "Voice Assist play music" << endl;
}

void Stereo::setVolume(int volume)
{
    this->volume = volume;
    cout << "Volume level: " << volume << endl;
}

const int Stereo::getVolume() { return volume; }

const int Stereo::getState() { return state; }